### DISCUSSION TITLE
I forget my 2fa code

### DISCUSSION THREAD

**Discussion by AgusXzz** [community/community#129765] [Code Security]
### Select Topic Area

General

### Body

How to generate new code

**Comment by vijayanjay** [comment]
Configure the 2FA again freshly by removing the old one.

**Reply by AgusXzz** [reply]
How?

**Comment by davevad93** [comment]
Hi @KilerBots ,

If you have lost your 2FA credentials you can use your recovery codes (which are stored in a `txt` file that you can download when you configure 2FA authentication), take a look at this from the docs:

- [Recovering your account if you lose your 2FA credentials](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials)

**Reply by D2nke** [reply]
Hi @davevad93, 

I lost my MFA code from my authentication app, I can't access the account because I don't have the recovery codes. I need to recover my account as I have commercial projects. Is there any way to recover? I also no longer have access to the computer I used to use.

**Reply by davevad93** [reply]
Hey @D2nke , in your case I think you should contact support by opening a ticket [here](https://support.github.com/contact/account) , select `Two-factor authentication recovery` or `Sign-in, verification or other account questions` and explain your issue.

**Comment by vishveshjain** [comment]
Regaining access to your GitHub account when you've forgotten your 2FA code depends on whether you have any backup methods set up. Here's how to proceed:

**If you have recovery codes:**

1. Unfortunately, GitHub won't be able to provide you with your forgotten 2FA code.

2. However, if you earlier saved recovery codes generated by GitHub, you can use them to regain access. These codes are meant for situations like this.

3. Locate the recovery codes you saved (hopefully a password manager!) and use them to log in to GitHub.

**If you don't have recovery codes:**

1. You'll need to initiate a 2FA recovery process through GitHub's account recovery.

2. Head to the password reset page on GitHub and enter your email address.

3. Click "Send password reset email" and follow the instructions in the email.

4. In the reset password process, you'll see an option like "Having problems?" Click that and choose "Start a 2FA recovery request".

5. Regaining access will depend on whether you have alternative verification methods set up on your account, such as:
   - A verified device you've previously used to log in to GitHub.
   - An SSH key.

If you have any of these set up, you can use them to verify your identity and regain access.

**If you don't have any recovery methods:**

Unfortunately, regaining access becomes more challenging. GitHub might offer limited options like unlinking your email address from the account. 

Here's the official GitHub documentation on recovering an account with lost 2FA credentials: [recovering your account if you lose your 2fa credentials ON GitHub docs.github.com]

If none of the above methods work, consider contacting GitHub support directly to explain your situation and see if they can offer further assistance.

**Comment by OCEANOFANYTHINGOFFICIAL** [comment]
If you've forgotten your GitHub 2FA code, you can follow these steps to recover your account:

1. **Recovery Codes**: When you set up 2FA on GitHub, you should have received a set of recovery codes. You can use one of these codes to regain access to your account. For more information on how to use recovery codes, refer to the [GitHub documentation](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials#using-a-recovery-code).

2. **SMS or Authentication App Backup**: If you set up a backup phone number or an alternative authentication app, you can use that to log in. Details on using your backup method are available [here](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials#using-an-sms-backup-code).

3. **Contact GitHub Support**: If you do not have your recovery codes and cannot access your backup methods, you will need to contact GitHub Support for help. You can start the process by visiting [this support page](https://support.github.com/request) and selecting the appropriate options for account recovery.

For more detailed instructions, please visit the official GitHub documentation on [recovering your account if you lose your 2FA credentials](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials).

**Comment by MattiaMDR** [comment]
these are very sensitive data, you must open a ticket here: https://support.github.com/contact/account
